File: src/main/java/pages/apidemos/logtextbox/LogTextBoxPage.java
package pages.apidemos.logtextbox;

import org.openqa.selenium.By;

import core.page.BasePage;
import io.appium.java_client.AppiumDriver;

public class LogTextBoxPage extends BasePage {
    private final By addButton = By.id("io.appium.android.apis:id/add");
    private final By logTextButtonArea = By.id("io.appium.android.apis:id/text");

    public LogTextBoxPage(AppiumDriver driver) {
        super(driver);
    }

    public LogTextBoxPage tapOnAddButton() {
        click(getElement(addButton));
        return this;
    }

    public String getLogText() {
        return getText(getElement(logTextButtonArea)).trim();
    }
}

File: src/main/java/pages/apidemos/logtextbox/LogTextBoxPage.kt
package pages.apidemos.logtextbox

import io.appium.java_client.AppiumDriver
import org.openqa.selenium.By
import org.openqa.selenium.WebElement

class LogTextBoxPage(private val driver: AppiumDriver<WebElement>) {
    private val addButton = By.id("io.appium.android.apis:id/add")
    private val logTextButtonArea = By.id("io.appium.android.apis:id/text")

    fun tapOnAddButton(): LogTextBoxPage {
        driver.findElement(addButton)?.click()
        return this
    }

    fun getLogText(): String {
        return driver.findElement(logTextButtonArea)!!.text!!.trim { it <= ' ' }
    }
}
